imports/test-utils/helpers/getFakeMongoCursor.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @name getFakeMongoCursor
+ * @param {String} collectionName - name of the collection
+ * @param {Any} results - results to be returned as part of the cursor
+ * @return {Object} fake cursor
+ */
+export default function getFakeMongoCursor(collectionName, results) {
```

Comment:
The same function is in `imports/plugins/core/graphql/server/resolvers/shop/Shop/administrators.test.js` and `imports/plugins/core/graphql/server/resolvers/account/Query/administrators.test.js`. Delete it from those files and import from here instead.

It would also be nice if this could replace the `mockCursor` I have in some test files. The problem is that those typically need the functions to be jest mocks rather than just plain functions. So for example the tests need:

```
limit: jest.fn().mockName("cursor.limit").mockReturnValue(cursor)
```

whereas for devserver we just need `() => cursor` because jest isn't loaded.

I think you could make this function a bit smarter to handle both use cases.

```
export default function getFakeMongoCursor(collectionName, results) {
  const cursor = {
    ns: `meteor.${collectionName}`,
    // ... etc. for all non-function props
  };
  const isTesting = typeof jest !== "undefined";

  if (isTesting) {
    cursor.limit = jest.fn().mockName("cursor.limit").mockReturnValue(cursor);
  } else {
    cursor.limit = () => cursor;
  }

  // and so on, for all props that are functions

  return cursor;
}
```

These are the places that have mock cursors for testing that could use this instead:

- imports/plugins/core/graphql/server/resolvers/util/applyPaginationToMongoCursor.test.js
- imports/plugins/core/graphql/server/resolvers/util/getCollectionFromCursor.test.js
- imports/plugins/core/graphql/server/resolvers/util/getPaginatedResponse.test.js

Do you mind giving that a try?

Reply:
looking into it